pkg/scheduler/framework/cycle_state.go

```diff
@@ -0,0 +1,85 @@
+package framework
+
+import (
+	"errors"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var (
+	// ErrNotFound is the not found error message.
+	ErrNotFound = errors.New("not found")
+)
+
+// StateData is a generic type for arbitrary data stored in CycleState.
+type StateData interface {
+	// Clone is an interface to make a copy of StateData. For performance reasons,
+	// clone should make shallow copies for members (e.g., slices or maps) that are not
+	Clone() StateData
+}
+
+// StateKey is the type of keys stored in CycleState.
+type StateKey string
+
+// CycleState provides a mechanism for plugins to store and retrieve arbitrary data.
+// StateData stored by one plugin can be read, altered, or deleted by another plugin.
+// CycleState does not provide any data protection, as all plugins are assumed to be
+// trusted.
+// Note: CycleState uses a sync.Map to back the storage. It's aimed to optimize for the "write once and read many times" scenarios.
+// It is the recommended pattern used in all in-tree plugins - plugin-specific state is written once in PreFilter/PreScore and afterwards read many times in Filter/Score.
+type CycleState struct {
+	// storage is keyed with StateKey, and valued with StateData.
+	storage sync.Map
+	// EnableEmptyWorkloadPropagation enable workload with replicas 0 to be propagated to member clusters.
+	EnableEmptyWorkloadPropagation bool
```

Comment:
What kind of information is appropriate to put in CycleState, for example, why not put `bindingSpec` and `bindingStatus` in CycleState as well?

Reply:
> why not put bindingSpec and bindingStatus in CycleState as well?

We should be like k8s.
```
SchedulePod func(ctx context.Context, fwk framework.Framework, state *framework.CycleState, pod *v1.Pod)
```

The pod is not in CycleState.

Reply:
This is not a clear rule, because I am wondering what to do with subsequent new input parameters (which may not exist).

Reply:
> new input parameters

Maybe can put it in CycleState, but i don't recommend put `bindingSpec` and `bindingStatus` in CycleState.